### PR TITLE
update test_builtin.py from cpython v3.11.2

### PR DESCRIPTION
**Pull request by JaeyoungAhn**
#4564 update test_builtin.py from cpython v3.11.2

### CODE REVIEW

File: Lib/test/test_builtin.py [hunk 1]
```diff
@@ -1226,7 +1364,7 @@ def test_open_default_encoding(self):
             os.environ.clear()
             os.environ.update(old_environ)
 
-    @unittest.skipIf(sys.platform == 'win32', 'TODO: RUSTPYTHON Windows')
+    @support.requires_subprocess()
```
Comment (DimitrisJim) [inline Lib/test/test_builtin.py:1367]: need to add back the `skipIf` here apparently, this test continues to fail on windows.

**Comment by DimitrisJim** [comment]
thanks! There's some additional regressions here which should be noted: `test_delattr`, `test_getattr`, `test_hasattr` and `test_setattr` should probably all be related. `test_type_name` is also failing now.

**Review by DimitrisJim — APPROVED** [review]
hopefully should be good to go now.

File: Lib/test/test_builtin.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def write(self, line):
         ('', ValueError),
         (' ', ValueError),
         ('  \t\t  ', ValueError),
-        # (str(br'\u0663\u0661\u0664 ','raw-unicode-escape'), 314), XXX RustPython
+        (str(br'\u0663\u0661\u0664 ','raw-unicode-escape'), 314),
```
Comment (youknowone) [inline Lib/test/test_builtin.py:97]: 👍

File: Lib/test/test_builtin.py [hunk 3]
```diff
@@ -1475,11 +1614,16 @@ def test_bug_27936(self):
             self.assertEqual(round(x, None), round(x))
             self.assertEqual(type(round(x, None)), type(round(x)))
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (youknowone) [inline Lib/test/test_builtin.py:1618]: fixing this one is looking like a good first issue

File: Lib/test/test_builtin.py [hunk 4]
```diff
@@ -519,10 +523,15 @@ def test_compile_async_generator(self):
         exec(co, glob)
         self.assertEqual(type(glob['ticker']()), AsyncGeneratorType)
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:527]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py [hunk 5]
```diff
@@ -748,11 +757,9 @@ def test_exec_globals(self):
         self.assertRaises(TypeError,
                           exec, code, {'__builtins__': 123})
 
-        # no __build_class__ function
-        code = compile("class A: pass", "", "exec")
-        self.assertRaisesRegex(NameError, "__build_class__ not found",
-                               exec, code, {'__builtins__': {}})
-
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:761]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py [hunk 6]
```diff
@@ -769,12 +776,55 @@ def __setitem__(self, key, value):
...
+                               exec, code, {'__builtins__': frozendict()})
+
         # read-only globals
         namespace = frozendict({})
         code = compile("x=1", "test", "exec")
         self.assertRaises(frozendict_error,
                           exec, code, namespace)
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:795]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py [hunk 7]
```diff
@@ -769,12 +776,55 @@ def __setitem__(self, key, value):
...
+        self.assertRaises(setonlyerror,
+                          exec, code, setonlydict({'globalname': 1}))
+
+        # builtins' `__getitem__` raises
+        code = compile("superglobal", "test", "exec")
+        self.assertRaises(setonlyerror, exec, code,
+                          {'__builtins__': setonlydict({'superglobal': 1})})
+
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:816]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py [hunk 8]
```diff
@@ -786,6 +836,86 @@ def test_exec_redirected(self):
         finally:
             sys.stdout = savestdout
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:840]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py [hunk 9]
```diff
@@ -817,19 +947,27 @@ def test_filter_pickle(self):
             f2 = filter(filter_char, "abcdeabcde")
             self.check_iter_pickle(f1, list(f2), proto)
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:951]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py [hunk 10]
```diff
@@ -817,19 +947,27 @@ def test_filter_pickle(self):
...
+        self.assertRaises(TypeError, getattr, sys)
+        msg = r"^attribute name must be string, not 'int'$"
+        self.assertRaisesRegex(TypeError, msg, getattr, sys, 1)
+        self.assertRaisesRegex(TypeError, msg, getattr, sys, 1, 'spam')
         self.assertRaises(AttributeError, getattr, sys, chr(sys.maxunicode))
         # unicode surrogates are not encodable to the default encoding (utf8)
         self.assertRaises(AttributeError, getattr, 1, "\uDAD1\uD51E")
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:964]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py (hunk 3 above)
Comment (fanninpm) [reply Lib/test/test_builtin.py:1618]: @DimitrisJim I believe this should have been broken out into a separate commit.

File: Lib/test/test_builtin.py [hunk 11]
```diff
@@ -2270,6 +2410,8 @@ def test_type_nokwargs(self):
         with self.assertRaises(TypeError):
             type('a', (), dict={})
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_builtin.py:2414]: @DimitrisJim I believe this should have been broken out into a separate commit.

**Pull request merged**